lib/babe/verify.go

```diff
@@ -219,16 +227,24 @@ func (v *VerificationManager) getVerifierInfo(epoch uint64) (*verifierInfo, erro
 	}, nil
 }
 
-func (v *VerificationManager) getConfigData(epoch uint64) (*types.ConfigData, error) {
+func (v *VerificationManager) getConfigData(epoch uint64, header *types.Header) (*types.ConfigData, error) {
 	for i := int(epoch); i >= 0; i-- {
 		has, err := v.epochState.HasConfigData(uint64(i))
 		if err != nil {
 			return nil, err
+		} else if !has {
+			continue
 		}
 
-		if has {
-			return v.epochState.GetConfigData(uint64(i))
+		configData, err := v.epochState.GetConfigData(uint64(i))
+		if errors.Is(chaindb.ErrKeyNotFound, err) {
+			configData, err = v.epochState.GetConfigDataForHeader(epoch, header)
+			if err != nil {
+				return nil, fmt.Errorf("config data was not found for epoch %d: %w", epoch, err)
+			}
 		}
+
+		return configData, err
```

Comment:
I don't think this is exactly right, in the case of unfinalized config data, then the loop will `continue` and not find it. I think you need to move the logic that checks `GetConfigData` and `GetConfigDataForHeader` to the top of the loop, and if both are nil, then continue. otherwise if one of them exists, then return it

Reply:
I have updated the `HasConfigData(uint64(i))` to check the in-memory next config data as well, so:

```
has, err := v.epochState.HasConfigData(uint64(i))
if !has { continue }
```
means that the config data is not in the database nor in the map otherwise, the data exists in one of them (database or map), I first check for the database if I get an `chaindb.ErrKeyNotFound` error then I search in the map.